lib/pug.js

```diff
@@ -8,14 +8,19 @@ const defaultConfig = { compile: {} }; // avoids key errors
 
 let hasConfig = true;
 try {
-  // require might fail
   require.resolve(configPath);
 } catch {
   hasConfig = false;
 }
 
 const config = hasConfig ? require(configPath) : defaultConfig;
 
+const hasProp = (obj, prop) => Object.prototype.hasOwnProperty.call(obj, prop);
+const invalidKeys = Object.keys(config).filter(k => !hasProp(defaultConfig, k));
+if (invalidKeys.length > 0) {
+  throw Error(`Unsupported PUG config keys: ${invalidKeys.join(', ')}`);
```

Comment:
TBH I would prefer just pass through every config keys to `pugjs`. And `pugjs` should validate whether it is valid option.

Reply:
Those keys are not PugJS keys, they're my own keys that map to a specific purpose (method) -- for example:
```js
{
  compile: {  // I validate this
      //... but not these (PugJS)
  }
}
```

I done this to prevent silent errors -- e.g. if someone set the `compileS` key, it would be silently ignored.